src/ctapipe/monitoring/calculator.py

```diff
@@ -0,0 +1,327 @@
+"""
+Definition of the ``StatisticsCalculator`` class, providing all steps needed to
+calculate the montoring data for the camera calibration.
+"""
+
+import numpy as np
+from astropy.table import Table, vstack
+
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import (
+    ComponentName,
+    Dict,
+    Float,
+    Int,
+    List,
+    TelescopeParameter,
+)
+from ctapipe.monitoring.aggregator import StatisticsAggregator
+from ctapipe.monitoring.outlier import OutlierDetector
+
+__all__ = [
+    "StatisticsCalculator",
+]
+
+
+class StatisticsCalculator(TelescopeComponent):
```

Comment:
One minor comment/question: since "monitoring" is more general (it's any time-series), this name is a bit generic, and it's not clear that the inputs/outputs of this are _pixel-wise_ monitoring information.  Maybe `PixelStatisticsCalculator` would be more clear?    Or is the intent here to eventually have this work for any monitoring data, not just pixel-wise?

Reply:
I agree with your suggestion. The `TelescopeComponent` is very tailored for _pixel-wise_ monitoring information at the moment. I'm not sure if all the functionality implemented for _pixel-wise_ monitoring data is needed for other regular monitoring data. There is also the [astropy aggregation](https://docs.astropy.org/en/stable/table/operations.html#aggregation) operations, which can be also utilized for rather slow-changing monitoring data.